recipes/sdl2/all/conanfile.py

```diff
@@ -0,0 +1,355 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.29.1"
+
+
+class SDL2Conan(ConanFile):
+    # TODO: When porting to CCI rename this package to SDL (without 2)
+    name = "sdl2"
+    description = "Access to audio, keyboard, mouse, joystick, and graphics hardware via OpenGL, Direct3D and Vulkan"
+    topics = ("sdl2", "audio", "keyboard", "graphics", "opengl")
+    url = "https://github.com/bincrafters/conan-sdl2"
+    homepage = "https://www.libsdl.org"
+    license = "Zlib"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = ["cmake", "pkg_config"]
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "directx": [True, False],
+        "alsa": [True, False],
+        "jack": [True, False],
+        "pulse": [True, False],
+        "sndio": [True, False],
+        "nas": [True, False],
+        "esd": [True, False],
+        "arts": [True, False],
+        "x11": [True, False],
+        "xcursor": [True, False],
+        "xinerama": [True, False],
+        "xinput": [True, False],
+        "xrandr": [True, False],
+        "xscrnsaver": [True, False],
+        "xshape": [True, False],
+        "xvm": [True, False],
+        "wayland": [True, False],
+        "directfb": [True, False],
+        "iconv": [True, False],
+        "video_rpi": [True, False],
+        "sdl2main": [True, False],
+        "opengl": [True, False],
+        "opengles": [True, False],
+        "vulkan": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "directx": True,
+        "alsa": True,
+        "jack": True,
+        "pulse": True,
+        "sndio": False,
+        "nas": True,
+        "esd": False,
+        "arts": False,
+        "x11": True,
+        "xcursor": True,
+        "xinerama": True,
+        "xinput": True,
+        "xrandr": True,
+        "xscrnsaver": True,
+        "xshape": True,
+        "xvm": True,
+        "wayland": False,
+        "directfb": False,
+        "iconv": True,
+        "video_rpi": False,
+        "sdl2main": True,
+        "opengl": True,
+        "opengles": True,
+        "vulkan": True
+    }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
```

Comment:
I would also remove iconv (maybe only for Visual Studio). In this branch https://github.com/libsdl-org/SDL/blob/889cebb7c20d4195e5d4ac344a2175f2490354cc/CMakeLists.txt#L779, iconv is never tested.